File: python/list.py
#!/usr/bin/python

import xmlrpclib

SATELLITE_URL = "https://rhn.redhat.com/rpc/api"
SATELLITE_LOGIN = "rhn-acccount"
SATELLITE_PASSWORD = "rhn-password"

client = xmlrpclib.Server(SATELLITE_URL, verbose=0)
key = client.auth.login(SATELLITE_LOGIN, SATELLITE_PASSWORD)
list = client.system.listUserSystems(key)

for group in list:
	sysname = group.get('name')
	sysid =  group.get('id')
	print '{0} - {1} ' .format (sysname, sysid)

client.auth.logout(key)

File: list.pl
#!/usr/bin/env perl


use strict;
use warnings;
use Frontier::Client;
use Data::Dumper;

my $server_url = 'https://rhn.redhat.com/rpc/api';

my $client = Frontier::Client->new('url' => $server_url);

my %config = do '/secret/rhn.config';

my $username = $config{'username'};
my $password = $config{'password'};

my $session = $client->call('auth.login', $username, $password);

my $systems = $client->call('system.list_user_systems', $session);

my %db;

for my $system (@$systems) {
	foreach my $key (keys %{$system}){
		print "$key : ${$system}{$key}\n";
		if($key eq 'id'){
			 my $entitlements = $client->call('system.getEntitlements', $session, ${$system}{$key});
			 if(!defined($$entitlements[0])){
				 print "entitlement: none\n";
				 $db{${$system}{$key}}=${$system}{'name'};
			 }else{
				print "entitlement: @$entitlements\n";
			}
		 }
	}
	print "=========================\n";
	#print Dumper($system);
}

foreach my $id (keys %db){
	print "$id : $db{$id}\n";
}

my @a= keys %db;

print "@a\n";
